### PR TITLE
Add libshaderc_combined to Windows builds.

### PR DESCRIPTION
**Pull request by AWoloszyn**
shaderc_combined.lib will now be built and tested when compiling
for MSVC or Ninja on Windows.

### CODE REVIEW

File: libshaderc/CMakeLists.txt [hunk 1]
```diff
@@ -24,16 +24,14 @@ shaderc_add_tests(
...
-  shaderc_combine_static_lib(shaderc_combined shaderc)
-  shaderc_add_tests(
-    TEST_PREFIX shaderc_combined
-    LINK_LIBS shaderc_combined ${CMAKE_THREAD_LIBS_INIT}
-    INCLUDE_DIRS include ${glslang_SOURCE_DIR}
-    TEST_NAMES
-      shaderc
-      shaderc_cpp)
-endif()
+# If we are not on windows, then create a single static library
```
Comment (dneto0) [inline libshaderc/CMakeLists.txt]: Comment is out of date.  Remove the "If we are not on windows" predicate.

File: cmake/utils.cmake [hunk 2]
```diff
@@ -113,45 +114,61 @@ macro(shaderc_get_transitive_libs target out_list)
...
+  if (MSVC)
     string(REPLACE ";" ">;$<TARGET_FILE:" temp_string "${all_libs}")
-    string(CONCAT all_libs_string "$<TARGET_FILE:" "${temp_string}" ">")
+    set(lib_target_list "$<TARGET_FILE:${temp_string}>")
 
-    add_custom_command(OUTPUT lib${new_target}.a
+    add_custom_command(OUTPUT ${libname}
       DEPENDS ${all_libs}
-      COMMAND libtool -static -o lib${new_target}.a ${all_libs_string})
+      COMMAND lib.exe ${lib_target_list} /OUT:${libname} /NOLOGO)
```
Comment (dneto0) [inline cmake/utils.cmake]: gotta love /NOLOGO

**Comment by dneto0** [comment]
Only nit is the out-of-date comment.
Once you fix that that you get a plus two from me.

**Comment by AWoloszyn** [comment]
Done!

File: cmake/utils.cmake [hunk 3]
```diff
@@ -91,7 +91,8 @@ function(shaderc_add_tests)
   endforeach()
 endfunction(shaderc_add_tests)
 
-# Finds all transitive static library dependencies of a given target.
+# Finds all transitive static library dependencies of a given target
+# including the target itself
```
Comment (dekimir) [inline cmake/utils.cmake]: "possibly including"?  Because the target is only included if it is, itself, a static library.

Comment (AWoloszyn) [inline cmake/utils.cmake]: Fixed.

**Comment by dekimir** [comment]
+1

**Comment by dneto0** [comment]
+2

**Pull request merged**